### ISSUE TITLE
Secret Variable exposed in debug when system.debug = true

### ISSUE THREAD

**Issue by pgunerud** [microsoft/azure-pipelines-task-lib#589]
### Environment
azure-pipelines-task-lib version: 2.9.3

### Issue Description
When a release has the variable system.debug = true and a task uses a secret variable that contains the string '%@' the secret variable will be exposed in the output like '##[debug]script=echo "Testing: %25@"'. 

Normal console.log works as expected and does not expose the secret variable.

This does not happen when using version 2.7.1 of azure-pipelines-task-lib.

### Expected behaviour
The secret value should not be exposed.  example:  '##[debug]script=echo "Testing: ***"'

### Actual behaviour
Secret value is exposed: example:  '##[debug]script=echo "Testing: %25@"

### Steps to reproduce
1. Create a task that uses the 2.9.3 aszure-pipelines-task-lib
2. Accept a string Input
3. Upload to Azure DevOps
4. Use Task in a release pipeline and pass a variable that has been marked as secret to the string input  Where the value of the secret variable contains %@.
5. Add variable system.debug = true to the pipeline variables
6. Run the release and view the output to see where getInput is called for the string input and it will display the secret variable.

### Logs
2019-11-19T00:43:18.1096717Z ##[debug]system.culture=en-US
2019-11-19T00:43:18.1096884Z ##[debug]failOnStderr=false
2019-11-19T00:43:18.1096972Z ##[debug]script=echo "Testing: %25@"

**Comment by stephenmichaelf** [comment]
Likely related to https://github.com/microsoft/azure-pipelines-task-lib/pull/543

**Comment by jtpetty** [comment]
This is fixed in version 2.164.1 of the agent. Closing this ticket.

**Comment by olidor** [comment]
Hi,

I've been able to reproduce using the following agent versions: **2.169.1** and **2.175.2**.

However, it only occurs for some secrets and only when the secret is fetched from a keyvault.
If the secret comes from a pipeline variable, then all is good and no secrets are exposed.

Bellow you can see some scenarios for which I was able to reproduce it or not.

```
Secret value				Behavior
%25this%25				not exposed
%25this%25 is no longer a secret	exposed
!25this%25 is no longer a secret	exposed
!25this!25 is no longer a secret	not exposed
!25this!25isnolongerasecret		not exposed
%25this%25isnolongerasecret		not exposed
!25this%25isnolongerasecret		exposed
```
Should I open a new incident?

Regards,
Olivian

p.s.: this is my task:

          - task: a8515ec8-7254-4ffd-912c-86772e2b5962@3
            inputs:
              targetFiles: '**/*Here.txt'
              encoding: 'auto'
              writeBOM: true
              actionOnMissing: 'warn'
              keepToken: false
              tokenPrefix: '#{'
              tokenSuffix: '}#'
              useLegacyPattern: false
              enableTelemetry: true
